### PR TITLE
opt-in to deduplication of cache populations

### PR DESCRIPTION
**Pull request by thundergolfer**
Reinstates #20 but makes it an optional feature of the `Cached` provider. 

This implementation requires a change to a `pub` fn and so is another minor ver bump, but the code is cleaner this way. I think an alternative is leaving the `pub` constructor alone and adding a builder that offers the functionality (seems unnecessary?)

### CODE REVIEW

**Comment by ekzhang** [comment]
Did we ever figure out why the deduplication was causing freezes?

**Comment by thundergolfer** [comment]
I did not, unfortunately. There's a bit of commentary in Slack `#development` yesterday. I set up a test blobnet instance running `0.2.9`, pointed a few dev workers at it, and couldn't repro the issue.

**Comment by ekzhang** [comment]
Why is it an optional feature?

**Comment by thundergolfer** [comment]
To not enable it for now on the server. https://github.com/modal-labs/blobnet/pull/28/files#diff-331b2ec889550798fe8bf57856543975a9ff1f1ae8b8b9c74aee2ea3b83623b6R80

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,2 @@
 [workspace]
 members = ["blobnet", "named-retry"]
-
-[profile.release]
-debug = 1
```
Comment (thundergolfer) [inline Cargo.toml:5]: From #26. Mistakenly added.

**Comment by thundergolfer** [comment]
Yep no worries at all. Will provide context, and I think "However, it sounds more like a temporary feature while we're debugging" is accurate.

#26 reverted the de-duping, because it was suspected of causing issues server-side in `blobnet.internal`. But we still need to use de-duping client-side, particularly for #25. Without de-duping we over-fetch by factor proportional to the ClientPool size, and this is really slow. 

I don't exact want to modify the public API, but seeing as I have I think I have to write docstrings explaining the use of the added functionality? 

If making this a temporary feature, then this commit avoids modifying the public constructor: [e3aa9a5](https://github.com/modal-labs/blobnet/pull/28/commits/e3aa9a59692b55af5a33ac74f1c859c2f05f5810).

**Comment by thundergolfer** [comment]
This PR may be superceded by https://github.com/modal-labs/blobnet/pull/29

**Pull request closed without merging**